syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.leveldb";
option java_outer_classname = "LevelDBProto";
option objc_class_prefix = "HLW";

package consistentHashing;

service ConsistentHashing {
  rpc Get (GetRequest) returns (GetResponse) {}
  rpc Put (PutRequest) returns (PutResponse) {}
  rpc AddNode (AddNodeRequest) returns (AddNodeResponse) {}
  rpc RemoveNode (PutRequest) returns (PutResponse) {}
  rpc Delete (DeleteRequest) returns (DeleteResponse){}
}

message DeleteRequest {
  string key = 1;
}

message DeleteResponse {
  string errormsg = 1;
}


message AddNodeRequest {
  string ip = 1;
}

message AddNodeResponse {
  string errormsg = 1;
}


message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 2;
}

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
  string errormsg = 1;
}